Add basic capability support

RockyTV opened the pull request:
The library now supports capability negotiation.
Spec: http://ircv3.net/specs/core/capability-negotiation-3.1.html

SirCmpwn commented on ChatSharp/IrcCapability.cs:
```diff
@@ -0,0 +1,31 @@
+        /// The state of the capability after negotiating with the server.
+        /// Disabled by default when the capability is created.
+        /// </summary>
+        public bool IsEnabled { get; set; }
```
Should this be internal set?

RockyTV commented:
Updated, set the properties to `internal set` and changed the return type of `Enabled` and `Disabled` to `IEnumerable<string>`. If the `Name` and `IsEnabled` properties are set to internal, there's not much reason to return a `IEnumerable<IrcCapability>`.

SirCmpwn commented:
Thanks!
